flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/MiniClusterITCase.java

```diff
@@ -159,6 +160,8 @@ private void setupAndRunHandleJobsWhenNotEnoughSlots(ScheduleMode scheduleMode)
     private void runHandleJobsWhenNotEnoughSlots(final JobGraph jobGraph) throws Exception {
         final Configuration configuration = getDefaultConfiguration();
         configuration.setLong(JobManagerOptions.SLOT_REQUEST_TIMEOUT, 100L);
+        configuration.setLong(
+                ResourceManagerOptions.STANDALONE_CLUSTER_STARTUP_PERIOD_TIME, 10000L);
```

Comment:
The test tests that batch slot requests are timed out on the `JobMaster` side via the slot request timeout. Since they are batch slot requests and there is at least on slot that can fulfill all slot requests, they won't be failed on the `ResourceManager` side. This, however, only works if the `TaskExecutor` registers within the `STANDALONE_CLUSTER_STARTUP_PERIOD_TIME` because otherwise there is no slot on the RM registered that can fulfill the slot requests.

Reply:
Thanks for the explanations! I got confused by:
- Not understanding properly how slot requests are handled for batch jobs
- Spending too much time in breakpoints while debugging, causing a different exception to be thrown :D (because timeout had elapsed in the meantime...)